auth/service.go

```diff
@@ -956,3 +961,122 @@ func DecodeDomainUserID(domainUserID string) (string, string) {
 		return "", ""
 	}
 }
+
+func (svc service) DeleteEntityPolicies(ctx context.Context, entityType, id string) (err error) {
+	switch entityType {
+	case ThingType:
+		// Remove policy of groups
+		req := PolicyReq{
+			SubjectType: GroupType,
+			Object:      id,
+			ObjectType:  ThingType,
+		}
+
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		// Remove policy from domain
+		req.SubjectType = DomainType
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		// Remove policy of users
+		req.SubjectType = UserType
+		if err := svc.DeletePolicy(ctx, req); err != nil {
+			return err
+		}
+
+		return nil
+	case UserType:
+		domainsPage, err := svc.domains.ListDomains(ctx, Page{SubjectID: id, Limit: defLimit})
+		if err != nil {
+			return err
+		}
+
+		if domainsPage.Total > defLimit {
+			for i := defLimit; i < int(domainsPage.Total); i += defLimit {
+				page := Page{SubjectID: id, Offset: uint64(i), Limit: defLimit}
+				dp, err := svc.domains.ListDomains(ctx, page)
+				if err != nil {
+					return err
+				}
+				domainsPage.Domains = append(domainsPage.Domains, dp.Domains...)
+			}
+		}
+
+		for _, domain := range domainsPage.Domains {
+			policy := PolicyReq{
+				Subject:     EncodeDomainUserID(domain.ID, id),
+				SubjectType: UserType,
+				Object:      domain.ID,
+				ObjectType:  DomainType,
+			}
+			if err := svc.agent.DeletePolicy(ctx, policy); err != nil {
+				return err
+			}
```

Comment:
This below snippet will delete all Thing, Group, Channel relation related to user id within the domain
```suggestion
			policy := PolicyReq{
				Subject:     EncodeDomainUserID(domain.ID, id),
				SubjectType: UserType,
				ObjectType:  ThingType,
			}
			if err := svc.agent.DeletePolicy(ctx, policy); err != nil {
				return err
			}
			policy = PolicyReq{
				Subject:     EncodeDomainUserID(domain.ID, id),
				SubjectType: UserType,
				ObjectType:  GroupType,
			}
			if err := svc.agent.DeletePolicy(ctx, policy); err != nil {
				return err
			}

			policy = PolicyReq{
				Subject:     EncodeDomainUserID(domain.ID, id),
				SubjectType: UserType,
				Object:      domain.ID,
				ObjectType:  DomainType,
			}
			if err := svc.agent.DeletePolicy(ctx, policy); err != nil {
				return err
			}
```